tests/integration/targets/iso_customize/tasks/iso_customize.yml

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2022, Ansible Project
+# Copyright (c) 2022, VMware, Inc. All Rights Reserved.
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Add a line to the file test02.cfg and make sure it succeed
+  ansible.builtin.lineinfile:
+    path: "{{ test_dir }}/test02.cfg"
+    regexp: "^test"
+    line: "test"
+
+- name: "Customize ISO file: add files and delete files"
+  community.general.iso_customize:
+    src_iso: "{{ test_dir }}/test.iso"
+    dest_iso: "{{ test_dir }}/{{ dest_iso_name }}"
+    delete_files:
+      - "/test01.cfg"
+      - "/test02.cfg"
+    add_files:
+      - src_file: "{{ test_dir }}/test01.cfg"
+        dest_file: "/preseed/ubuntu.seed"
+      - src_file: "{{ test_dir }}/test02.cfg"
+        dest_file: "/test02.cfg"
+
+# Only need to check the files is deleted/added as the requested modifications 
+# in one type of OS(take MacOS here)
```

Comment:
Why only in one type of OS? Why not in all? This makes no sense to me.

Reply:
@felixfontein 
Thanks for the question.

As I understanding, when we check if the files is changed as expected or not in test_cusotmized.iso, we extract the iso and check the content of changed file. It will be enough we do this in one type of OS distribution (such as in my local MAC).
I think I don't need to extract the iso and check the content in all OS distributions.  I believe the extracted content of the ISO will be same whenever you extract it in Mac or Ubuntu or others.

In fact, at firstly I use the common module "ansible.posix.mount" in order to check the content of test_customized.iso.  But it reports error when i run main.yml in my local MAC.  
If I use different command to mount ISO for different OS, I may need to figure out which command i should use in different OS. (such as freeBSD ..). It may consume a lot of my time and I think i don't have to do it.

Correct me if I am wrong.

Best regards
Yuhua Zou

Reply:
In that case you should write the tests that it is easy to support more systems. Right now it is not: you have to change every condition `ansible_distribution == "MacOSX"`.

How about moving the mount / unmount step to a different task file, and wrap all tests into a global `block:` with one single `when:` statement.

Reply:
> wrap all tests into a global `block:` with one single `when:` statement.

Thanks. change it.